src/firefly/js/ui/RadiusInputFields.jsx

```diff
@@ -0,0 +1,205 @@
+
+/*
+ * remove trailing zero from toFixed result
+ */
+function toMaxFixed(floatNum, digits) {
+    return parseFloat(floatNum.toFixed(digits));
+}
+
+// in string format, size in degree
+var sizeToDeg = (size, unit) => (
+    (size)? toMaxFixed(convertAngle(((unit) ? unit :'deg'), 'deg', size), DECDIGIT).toString() : ''
+);
+
+// in string format, size from degree to any other unit
+var sizeFromDeg= (sizeDeg, unit) => (
+    (sizeDeg)? toMaxFixed(convertAngle('deg', ((unit) ? unit :'deg'), sizeDeg), DECDIGIT).toString() : ''
+);
+
+// validate size (in degree) within the range
+var isSizeValid = (sizeDeg,  min, max) => (
+    (sizeDeg && Validate.floatRange(min, max, 1, 'value of radius size in degree', sizeDeg).valid) ? true : false
+);
+
+
+function getProps(params, fireValueChange) {
+
+    var unit = params.unit ? params.unit : 'deg';
+    var valid = isSizeValid(params.value, params.min, params.max);
+    var value = (valid&&params.value) ? params.value  : '';
+    var displayValue = params.displayValue ? params.displayValue : sizeFromDeg(value, unit);
+
+    return Object.assign({}, params,
+        {
+            onChange: (ev) => handleOnChange(ev, params, fireValueChange),
+            unit,
+            displayValue,
+            value,
+            valid
+        });
+}
+
+function handleOnChange(ev, params, fireValueChange) {
+     var sizeInfo = Object.assign({}, {displayValue: '', unit: 'deg', value: '', valid: true}, ev.sizeInfo);
+     var {displayValue, unit, value, valid} = sizeInfo;
+
+     fireValueChange({
+         feedback: valid ? invalidSizeMsg: '',
+         displayValue,
+         unit,
+         value,
+         valid
+     });
+}
+
+const propTypes={
+    label:       PropTypes.string,
+    labelWidth:  PropTypes.number,
+    value:       PropTypes.string.required,
+    unit:        PropTypes.string,
+    min:         PropTypes.number.isRequired,
+    max:         PropTypes.number.isRequired
+};
+
+/**
+ * @param {string} value size value in degree
+ * @param {string} displayValue size value displayed
+ * @param {string} unit selected unit
+ * @param {bool}   valid validation of size value in degree
+ *
+ */
+
+class RadiusInputFieldView extends Component {
+    constructor(props) {
+        super(props);
+
+        this.updateState(props);
+
+        this.onSizeChange = this.onSizeChange.bind(this);
+        this.onUnitChange = this.onUnitChange.bind(this);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.updateState(nextProps);
+    }
+
+    updateState(props) {
+        var unit = props.unit ? props.unit : 'deg';
+        var valid = isSizeValid(props.value, props.min, props.max);
+        var value = (valid&&props.value) ? props.value  : '';
+        var displayValue = props.displayValue ? props.displayValue : sizeFromDeg(value, unit);
+
+        this.state = {
+            value,
+            unit,
+            displayValue,
+            valid
+        };
```

Comment:
This is not the right way to update state in react.  I am surprised it even works.  in the constructor `this.state= {...}` is correct.  However anywhere else you should call `this.setState({...})`